theano/gpuarray/basic_ops.py

```diff
@@ -1613,21 +1613,24 @@ def grad(self, inp, grads):
 
     def gpu_kernels(self, node, name):
         code = """
-KERNEL void eye(GLOBAL_MEM %(ctype)s *a, ga_size n, ga_size m) {
-    ga_size nb = n < m ? n : m;
+KERNEL void eye(GLOBAL_MEM %(ctype)s *a, ga_size n, ga_size m, ga_int k) {
+    ga_int col_off = max(k, (ga_int) 0);
+    ga_int row_off = -min(k, (ga_int) 0);
+    ga_size nb = (ga_size) min(n - row_off, m - col_off);
+    nb = max(nb, (ga_size) 0);
     for (ga_size i = LID_0; i < nb; i += LDIM_0) {
-        a[i*m + i] = %(write_a)s(1);
+        a[(i + row_off)*m + i + col_off] = %(write_a)s(1);
     }
 }""" % dict(ctype=pygpu.gpuarray.dtype_to_ctype(self.dtype),
             name=name, write_a=write_w(self.dtype))
         return [Kernel(
                 code=code, name="eye",
-                params=[gpuarray.GpuArray, gpuarray.SIZE, gpuarray.SIZE],
+                params=[gpuarray.GpuArray, gpuarray.SIZE, gpuarray.SIZE, np.int32],
```

Comment:
You need to indicate the type of ```k``` as ```'int32'```. Take a look at [this example](https://github.com/Theano/Theano/blob/99e32f41ec92fe2d123c7770074b99674d9d5073/theano/gpuarray/elemwise.py#L2626).